### DISCUSSION TITLE
I'm need al selected rows livewire-tabels

### DISCUSSION THREAD

**Discussion by Giovanni-Petrella** [rappasoft/laravel-livewire-tables#1805] [Q&A]
I'm need al selected rows data but not exist on livewire-tables 2 $this->selectedRowsQuery()->get(), how can I do it?

**Comment by lrljoe** [comment]
What're you trying to achieve?  Do you want to just get  a list of Models that have been selected?  Or something more?

**Comment by Giovanni-Petrella** [comment]
I need the selected models to filter them based on an attribute and then pass the list of filtered ids to a modal that needs to do some processing on the models that have certain characteristics. In short, I would like to avoid passing all the selected ones but only those that have a certain attribute, I am migrating all my datatables from livewire-datatables to your livewire-tablee.

**Reply by lrljoe** [reply]
Are you using Version 2, or Version 3?

Version 2 is on Livewire 2, and Livewire 2 isn't actively maintained anymore.
Version 3 is on Livewire 3 and very actively maintained.

**Comment by Giovanni-Petrella** [comment]
So I'm at version 2 because my software is laravel 9 livewire 2 and livewire-datatables and I'm migrating to your great extension of very complex tables to play all the things I had on the old extension. So I need to reproduce all the features I had and this is one that I can't make work as I need as I need to have the data of the selected rows and do some checks first to avoid passing too much data to the modal that has to use the data for other things. I specify that I do this migration precisely in order to then migrate laravel to version 11, livewire at 3 and yours at 3 but I can't do otherwise before I have to have everything ready and compatible.

**Reply by lrljoe** [reply]
Okay, when you say selected rows, are you talking Bulk Actions?

OR are you wanting to just get all of the records that are returned based on the user's searches/filters?

For Bulk Actions, you can retrieve the Primary Keys/IDs for the selected rows via
```php
$selectedRows = $this->getSelected();
```
Now, I'd advise with Livewire 2 that you pass the IDs into your modal, rather than trying to pass the actual rows, and then in your modal pull back the data that you need from the database.

In Version 2, at the point that the Bulk Action is running, it just has the list of IDs for the items you've selected.

If you need to pass actual data across, then you could just do this in your Bulk Action:
```php
$items = YourModelName::select(['id', 'name', 'other_field'])->whereIn('id', $this->getSelected())->get();
```
And that'd execute a fresh query with the IDs that are in the "selected" array.

Because of the way that everything is paginated etc, and only relevant fields retrieved, the Table only queries the database for the data it needs for the next render, so you'd have to pull the data back regardless, and you may as well do it in your Modal (so just passing a list of IDs to it).

**Comment by Giovanni-Petrella** [comment]
Ok thanks, in fact I had already done as you advised on Bulk-Action me I just wanted to optimize as I had done previously by doing a check before passing the ids in modal form, this was possible because in that datable there is a function that provides me with all the data of the rows selected as I thought the one mentioned by you would do: $this->selectedRowsQuery() 
Just to show how I did it before and what control I did, I quote the code I had previously:
```php
//recupero degli ids degli shipments con allegati selezionati nell'ordine di visualizzazione al momento della richiesta
        $selected = $this->getExportResultsSet()->filter(function ($row) {
            return $row['callback_attachments_count'] > 0;
        })->map(function ($printable) {
            return $printable['Id'];
        })->toArray();
        $this->emit('showMassPrint', $selected);
``` 

here $row ['callback_attachments_count'] is the filter I want to apply to filter the ids before sending them to modal and the function that provides me with the selected data which also includes the attribute to filter with is $this->getExportResultsSet(). 
Anyway, thanks, after I migrate to version 3 of your extension I'll see if I can optimize this.
